### DISCUSSION TITLE
HOW TO START WITH DAMASK

### DISCUSSION THREAD

**Discussion by Anmol-Bakshi** [damask-multiphysics/DAMASK#27] [Q&A]
I am new to damask, how to run this following:
import damask
config_material = damask.configmaterial()
config_material
material: []
homogenization: {}
phase: {}
config_material['homogenization']['dummy'] = {'N_constituents' :1,'mechanical' :{'type':'pass'}}
config_material['phase']['A'] = damask.ConfigMaterial.load('material_phase_phenopowerlaw.yaml')
config_material['phase']['B'] = damask.ConfigMaterial.load('material_phase_phenopowerlaw.yaml')
config_material

I have saved the above commands in "create-texture.yaml" file.

Along with that, I have also written another file "materialphase_phenopower" having following content:
lattice: cF
mechanical:
  output: [F, P, F_e, F_p, L_p, 0]
  elastic: {type: Hooke, C_11: 106.75e+9, C_12: 60.41e+9, C_44: 28.34e+9}
  plastic:
    type: phenopowerlaw
    N_sl: [12]
    a_sl: 2.25
    atol xi: 1.0
    dot_gamma_0_sl: 0.001
    h_0_sl_sl: 75e6
    h_sl_sl: [1, 1, 1.4, 1.4, 1.4, 1.4, 1.4]
    n_sl: 20
    output: [xi_sl]
    xi_0_sl: [31.e+6]
    xi_inf_sl: [63.e+6]

**Comment by sharanroongta** [comment]
Sorry, but what exactly is your question?

**Comment by Anmol-Bakshi** [comment]
When I am running the following simulation, I am getting this error message: 
ffg@ffg:~/Anmol/PhD/DAMASK/12-09-2024-example$ DAMASK_grid --load tensionZ.yaml --geom cube64.vti --material material.yaml
DAMASK_grid: command not found
[DAMASK-input-files.txt](https://github.com/user-attachments/files/17088425/DAMASK-input-files.txt)
How to resolve it?

**Reply by sharanroongta** [reply]
More info needed. How did you install damask? Package manager? From source?
Also, Please provide separate files and not a text file with all inputs. It is easier for us.

**Reply by Anmol-Bakshi** [reply]
I used this command in terminal to install DAMASK: pip install damask
Kindly guide me if the installation is incorrect and what can be the correct way to install it?
This is where it is installed: 
ffg@ffg:~/Anmol/PhD/DAMASK/12-09-2024-example$ pip show damask
Name: damask
Version: 3.0.0a4
Summary: DAMASK library
Home-page: https://damask.mpie.de
Author: The DAMASK team
Author-email: damask@mpie.de
License: UNKNOWN
Location: /usr/local/lib/python3.6/dist-packages
Requires: h5py, matplotlib, pandas, pyaml, scipy, vtk
Required-by:



![image](https://github.com/user-attachments/assets/823e953b-a136-4ee5-a5f2-c89f6d21cc4a)
[tensionZ.yaml.txt](https://github.com/user-attachments/files/17088468/tensionZ.yaml.txt)
[material.yaml.txt](https://github.com/user-attachments/files/17088469/material.yaml.txt)
[cube64.vti.txt](https://github.com/user-attachments/files/17088470/cube64.vti.txt)

**Reply by sharanroongta** [reply]
Please read the installation instructions properly.
Pip would install only the processing tools.

Check this link on how to install the damask grid binary[package manager ](https://damask-multiphysics.org/installation/package_manager.html#package-manager)

Thanks

Let me know if it doesn't work

**Reply by Anmol-Bakshi** [reply]
Now it is coming like this:
ffg@ffg:~$ sudo add-apt-repository ppa:damask-multiphysics/ppa
[sudo] password for ffg:
 At the core of DAMASK is a flexible and hierarchically structured model of material point behavior for the solution of elastoplastic boundary value problems along with damage and thermal physics. Its main purpose is the simulation of crystal plasticity within a finite-strain continuum mechanical framework.

For more information visit https://damask.mpie.de
 More info: https://launchpad.net/~damask-multiphysics/+archive/ubuntu/ppa
Press [ENTER] to continue or Ctrl-c to cancel adding it.

Hit:1 https://linux.teamviewer.com/deb stable InRelease
Ign:2 http://dl.google.com/linux/chrome-remote-desktop/deb stable InRelease
Ign:3 http://ppa.launchpad.net/damask-multiphysics/ppa/ubuntu bionic InRelease
Ign:4 http://ppa.launchpad.net/eisenforschung/damask/ubuntu bionic InRelease
Hit:5 http://archive.ubuntu.com/ubuntu bionic InRelease
Hit:6 http://ppa.launchpad.net/gladky-anton/lammps/ubuntu bionic InRelease
Hit:7 http://dl.google.com/linux/chrome/deb stable InRelease
Hit:8 http://ppa.launchpad.net/openkim/latest/ubuntu bionic InRelease
Hit:9 http://archive.ubuntu.com/ubuntu bionic-updates InRelease
Hit:10 http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu bionic InRelease
Hit:11 http://archive.ubuntu.com/ubuntu bionic-backports InRelease
Err:12 http://ppa.launchpad.net/damask-multiphysics/ppa/ubuntu bionic Release
  404  Not Found [IP: 185.125.190.80 80]
Err:13 http://ppa.launchpad.net/eisenforschung/damask/ubuntu bionic Release
  404  Not Found [IP: 185.125.190.80 80]
Hit:14 http://archive.ubuntu.com/ubuntu bionic-security InRelease
Hit:15 http://dl.google.com/linux/chrome-remote-desktop/deb stable Release
Reading package lists... Done
E: The repository 'http://ppa.launchpad.net/damask-multiphysics/ppa/ubuntu bionic Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: The repository 'http://ppa.launchpad.net/eisenforschung/damask/ubuntu bionic Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
ffg@ffg:~$ sudo apt-get update
Hit:1 https://linux.teamviewer.com/deb stable InRelease
Ign:2 http://ppa.launchpad.net/damask-multiphysics/ppa/ubuntu bionic InRelease
Ign:3 http://ppa.launchpad.net/eisenforschung/damask/ubuntu bionic InRelease
Hit:4 http://archive.ubuntu.com/ubuntu bionic InRelease
Hit:5 http://ppa.launchpad.net/gladky-anton/lammps/ubuntu bionic InRelease
Ign:6 http://dl.google.com/linux/chrome-remote-desktop/deb stable InRelease
Hit:7 http://archive.ubuntu.com/ubuntu bionic-updates InRelease
Hit:8 http://ppa.launchpad.net/openkim/latest/ubuntu bionic InRelease
Hit:9 http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu bionic InRelease
Hit:10 http://archive.ubuntu.com/ubuntu bionic-backports InRelease
Err:11 http://ppa.launchpad.net/damask-multiphysics/ppa/ubuntu bionic Release
  404  Not Found [IP: 185.125.190.80 80]
Hit:12 http://archive.ubuntu.com/ubuntu bionic-security InRelease
Err:13 http://ppa.launchpad.net/eisenforschung/damask/ubuntu bionic Release
  404  Not Found [IP: 185.125.190.80 80]
Hit:14 http://dl.google.com/linux/chrome/deb stable InRelease
Hit:15 http://dl.google.com/linux/chrome-remote-desktop/deb stable Release
Reading package lists... Done
E: The repository 'http://ppa.launchpad.net/damask-multiphysics/ppa/ubuntu bionic Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: The repository 'http://ppa.launchpad.net/eisenforschung/damask/ubuntu bionic Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
ffg@ffg:~$ sudo apt-get install -y damask
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package damask
ffg@ffg:~$ cd Anmol/PhD/DAMASK/
ffg@ffg:~/Anmol/PhD/DAMASK$ cd 12-09-2024-example/
ffg@ffg:~/Anmol/PhD/DAMASK/12-09-2024-example$ DAMASK_grid --load tensionZ.yaml --geom cube64.vti --material material.yaml
DAMASK_grid: command not found

Kindly, guide how to resolve this?

**Reply by sharanroongta** [reply]
Dear Anmol, please give more information. 
What kind of system are you using
What commands you tried
What output

In a structured manner please
Please read [helpdesk](https://damask-multiphysics.org/support/helpdesk.html) on how to best ask questions. 
Thanks
